pipeline-scripts/release.groovy

```diff
@@ -460,4 +460,235 @@ def signArtifacts(Map signingParams) {
+
+                    /**
+                     * @param name - The release name
+                     * @param version - a list of versions in a channel
+                     * @return Returns true if name or name+<arch> exists in list
+                     */
+                    isInChannel = { name, versions  ->
+                        for ( String version : versions ) {
+                            if ( version.startsWith( name + "+") || version.equals(name) ) {
+                                return true
+                            }
+                        }
+                        return false
+                    }
+
+                    addToChannel = !isInChannel(releaseName, channelYaml.get('versions', []))
+
+                    upgradeChannelYaml = [ name: upgradeChannel, versions: [] ]
+                    releasesForUpgradeChannel = []
+                    if (fileExists(upgradeChannelFile)) {
+                        upgradeChannelYaml = readYaml(file: upgradeChannelFile)
+                        upgradeChannelVersions = upgradeChannelYaml.get('versions', [])
+
+                        // at least one version must be present & make sure that releaseName is not already there
+                        if ( upgradeChannelVersions && !isInChannel(releaseName, upgradeChannelVersions) ) {
+                            /**
+                             * We need find the most recent version of 4.minor+1 and determine which
+                             * architectures it supports. If it only supports x86_64, the OTA team
+                             * only wants us to PR for 4.y+1 with a CPU suffix which limits upgrades
+                             * to 4.y+1 from this release's x86_64 image.
+                             */
+                            def spacedVersions = upgradeChannelVersions.join(' ')
+                            // Use sort -V to handle semver sorting of versions and choose the last
+                            def lastNextVer = sh(returnStdout: true, script: "echo ${spacedVersions} | tr ' ' '\\n' | grep ${major}[.]${minorNext}[.] | sort -V | tail -n 1").trim()
+                            if ( lastNextVer ) {
+
+                                /**
+                                 * @param name - The name of the release to look for
+                                 * @param arch - The architecture you want to know if the release is defined for
+                                 * @return Returns true if the release specified has support for the specified architecture
+                                 */
+                                def existsForArch = { name, arch ->
+                                    def suffix = getArchSuffix(arch)
+                                    // If there is an imagestream in the arch's api.ci namespace, it was built for that arch by ART.
+                                    // Check for both is/release and is/release-<ach> (the latter being an older style of release naming)
```

Comment:
```suggestion
                                    // Check for both is/release and is/release-<arch> (the latter being an older style of release naming)
```
... which we will never use again, so I'm wondering if it really should be looking for both.

Reply:
Just my personal paranoia shining through.. that we will switch back for some reason. Or run one accidentally and just run with it because we are out of time.